--- pallets/transaction-payment/.tmp/original/Cargo.toml
+++ pallets/transaction-payment/.tmp/migrate/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
-name = "pallet-transaction-payment"
+name = "rpallet-transaction-payment"
 version = "4.0.0-dev"
-authors = ["Parity Technologies <admin@parity.io>"]
+authors = ["Parity Technologies <admin@parity.io>", "Rio Team<dev@riodefi.com>"]
 edition = "2021"
 homepage = "https://substrate.io"
 license = "Apache-2.0"
