--- node/relaychain-rio-cli/Cargo.toml.tmp
+++ node/relaychain-rio-cli/Cargo.toml.tmp.migrate
@@ -1,5 +1,5 @@
 [package]
-name = "polkadot-cli"
+name = "relaychain-rio-cli"
 version = "0.9.22"
 authors = ["Parity Technologies <admin@parity.io>"]
 description = "Polkadot Relay-chain Client Node"
@@ -20,8 +20,10 @@
 futures = "0.3.21"
 pyro = { package = "pyroscope", version = "0.3.1", optional = true }
 
-service = { package = "polkadot-service", git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", default-features = false, optional = true }
-polkadot-client = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
+service = { package = "relaychain-rio-service", path = "../relaychain-rio-service", default-features = false, optional = true }
+#service = { package = "polkadot-service", git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", default-features = false, optional = true }
+polkadot-client = { package = "relaychain-rio-client", path = "../relaychain-rio-client", optional = true }
+#polkadot-client = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
 polkadot-node-core-pvf = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
 polkadot-performance-test = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
 
