--- runtime/.tmp/original/Cargo.toml
+++ runtime/.tmp/migrate/Cargo.toml
@@ -1,5 +1,5 @@
 [package]
-name = "parachain-template-runtime"
+name = "parachain-rio-runtime"
 version = "0.1.0"
 authors = ["Anonymous"]
 edition = "2021"
@@ -22,9 +22,6 @@
 serde = { version = "1.0.137", optional = true, features = ["derive"] }
 smallvec = "1.6.1"
 
-# Local
-pallet-template = { git = "https://github.com/substrate-developer-hub/substrate-parachain-template", rev = "polkadot-v0.9.24", default-features = false }
-
 # Substrate
 frame-benchmarking = { git = "https://github.com/paritytech/substrate", default-features = false, optional = true, branch = "polkadot-v0.9.24" }
 frame-executive = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.24" }
@@ -39,8 +36,8 @@
 pallet-session = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.24" }
 pallet-sudo = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.24" }
 pallet-timestamp = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.24" }
-pallet-transaction-payment = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.24" }
-pallet-transaction-payment-rpc-runtime-api = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.24" }
+pallet-transaction-payment = { package = "rpallet-transaction-payment", path = "../pallets/transaction-payment", default-features = false }
+pallet-transaction-payment-rpc-runtime-api = { package = "rpallet-transaction-payment-rpc-runtime-api", path = "../pallets/transaction-payment/rpc/runtime-api", default-features = false }
 sp-api = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.24" }
 sp-block-builder = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.24" }
 sp-consensus-aura = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.24" }
@@ -57,7 +54,7 @@
 # Polkadot
 pallet-xcm = { git = "https://github.com/paritytech/polkadot", default-features = false, branch = "release-v0.9.24" }
 polkadot-parachain = { git = "https://github.com/paritytech/polkadot", default-features = false, branch = "release-v0.9.24" }
-polkadot-runtime-common = { git = "https://github.com/paritytech/polkadot", default-features = false, branch = "release-v0.9.24" }
+polkadot-runtime-common = { package = "rio-runtime-common", path = "common-rio", default-features = false }
 xcm = { git = "https://github.com/paritytech/polkadot", default-features = false, branch = "release-v0.9.24" }
 xcm-builder = { git = "https://github.com/paritytech/polkadot", default-features = false, branch = "release-v0.9.24" }
 xcm-executor = { git = "https://github.com/paritytech/polkadot", default-features = false, branch = "release-v0.9.24" }
@@ -75,6 +72,33 @@
 pallet-collator-selection = { git = "https://github.com/paritytech/cumulus", default-features = false, branch = "polkadot-v0.9.24" }
 parachain-info = { git = "https://github.com/paritytech/cumulus", default-features = false, branch = "polkadot-v0.9.24" }
 
+# Frontier
+fp-rpc = { path = "../frontier/primitives/rpc", default-features = false }
+fp-self-contained = { path = "../frontier/primitives/self-contained", default-features = false }
+pallet-base-fee = { path = "../frontier/frame/base-fee", default-features = false }
+pallet-dynamic-fee = { path = "../frontier/frame/dynamic-fee", default-features = false }
+pallet-ethereum = { path = "../frontier/frame/ethereum", default-features = false }
+pallet-evm = { path = "../frontier/frame/evm", default-features = false }
+pallet-evm-precompile-dispatch = { path = "../frontier/frame/evm/precompile/dispatch", default-features = false }
+pallet-evm-precompile-modexp = { path = "../frontier/frame/evm/precompile/modexp", default-features = false }
+pallet-evm-precompile-sha3fips = { path = "../frontier/frame/evm/precompile/sha3fips", default-features = false }
+pallet-evm-precompile-simple = { path = "../frontier/frame/evm/precompile/simple", default-features = false }
+pallet-hotfix-sufficients = { path = "../frontier/frame/hotfix-sufficients", default-features = false }
+
+pallet-evm-precompile-assets-erc20 = { path = "../precompiles/assets-erc20", default-features = false }
+
+# Orml
+orml-currencies = { git = "https://github.com/open-web3-stack/open-runtime-module-library.git", rev = "27e3272d322dcdd915f0fc2002032e8d53a46523", version = "0.4.1-dev", default-features = false }
+orml-traits = { git = "https://github.com/open-web3-stack/open-runtime-module-library", rev = "27e3272d322dcdd915f0fc2002032e8d53a46523", version = "0.4.1-dev", default-features = false }
+rio-gateway-rpc-runtime-api = { path = "../pallets/gateway/rpc/runtime-api", default-features = false }
+# Rio
+rp-base = { path = "../primitives/base", default-features = false }
+rp-protocol = { path = "../primitives/protocol", default-features = false }
+rpallet-assets = { path = "../pallets/assets", default-features = false }
+rpallet-assets-ext = { path = "../pallets/assets-ext", default-features = false }
+rpallet-gateway = { path = "../pallets/gateway", default-features = false }
+rpallet-staking-pool = { path = "../pallets/staking-pool", default-features = false }
+
 [features]
 default = [
   "std",
@@ -102,7 +126,6 @@
   "pallet-collator-selection/std",
   "pallet-session/std",
   "pallet-sudo/std",
-  "pallet-template/std",
   "pallet-timestamp/std",
   "pallet-transaction-payment-rpc-runtime-api/std",
   "pallet-transaction-payment/std",
@@ -125,6 +148,27 @@
   "xcm-builder/std",
   "xcm-executor/std",
   "xcm/std",
+  "orml-traits/std",
+  "orml-currencies/std",
+  "rpallet-assets/std",
+  "rpallet-assets-ext/std",
+  "rpallet-staking-pool/std",
+  "rpallet-gateway/std",
+  "rio-gateway-rpc-runtime-api/std",
+  "rp-base/std",
+  "rp-protocol/std",
+  "fp-rpc/std",
+  "fp-self-contained/std",
+  "pallet-base-fee/std",
+  "pallet-dynamic-fee/std",
+  "pallet-ethereum/std",
+  "pallet-evm/std",
+  "pallet-evm-precompile-modexp/std",
+  "pallet-evm-precompile-sha3fips/std",
+  "pallet-evm-precompile-simple/std",
+  "pallet-evm-precompile-dispatch/std",
+  "pallet-hotfix-sufficients/std",
+  "pallet-evm-precompile-assets-erc20/std",
 ]
 
 runtime-benchmarks = [
@@ -135,7 +179,6 @@
   "frame-system/runtime-benchmarks",
   "pallet-balances/runtime-benchmarks",
   "pallet-collator-selection/runtime-benchmarks",
-  "pallet-template/runtime-benchmarks",
   "pallet-timestamp/runtime-benchmarks",
   "pallet-xcm/runtime-benchmarks",
   "sp-runtime/runtime-benchmarks",
@@ -148,3 +191,6 @@
   "frame-executive/try-runtime",
   "frame-try-runtime",
 ]
+
+# Set timing constants (e.g. session period) to faster versions to speed up testing.
+fast-runtime = []
