--- node/relaychain-rio-client/.tmp/original/Cargo.toml
+++ node/relaychain-rio-client/.tmp/migrate/Cargo.toml
@@ -1,5 +1,5 @@
 [package]
-name = "polkadot-client"
+name = "relaychain-rio-client"
 version = "0.9.24"
 authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2021"
@@ -39,7 +39,7 @@
 
 # Polkadot Runtimes
 kusama-runtime = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.24", optional = true }
-polkadot-runtime = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.24", optional = true }
+polkadot-runtime = { package = "relaychain-rio-runtime", path = "../../runtime/relaychain-rio", optional = true }
 rococo-runtime = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.24", optional = true }
 westend-runtime = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.24", optional = true }
 
@@ -54,6 +54,8 @@
 kusama = ["kusama-runtime"]
 rococo = ["rococo-runtime"]
 westend = ["westend-runtime"]
+fast-runtime = ["polkadot-runtime/fast-runtime"]
+rio-testnet = ["polkadot-runtime/rio-testnet"]
 runtime-metrics = [
   "rococo-runtime/runtime-metrics",
   "kusama-runtime/runtime-metrics",
