--- node/relaychain-rio-cli/src/command.rs.tmp
+++ node/relaychain-rio-cli/src/command.rs
@@ -82,7 +82,6 @@
 			id
 		};
 		Ok(match id {
-			"kusama" => Box::new(service::chain_spec::kusama_config()?),
 			#[cfg(feature = "kusama-native")]
 			"kusama-dev" => Box::new(service::chain_spec::kusama_development_config()?),
 			#[cfg(feature = "kusama-native")]
@@ -92,14 +91,12 @@
 			#[cfg(not(feature = "kusama-native"))]
 			name if name.starts_with("kusama-") && !name.ends_with(".json") =>
 				Err(format!("`{}` only supported with `kusama-native` feature enabled.", name))?,
-			"polkadot" => Box::new(service::chain_spec::polkadot_config()?),
 			#[cfg(feature = "polkadot-native")]
 			"polkadot-dev" | "dev" => Box::new(service::chain_spec::polkadot_development_config()?),
 			#[cfg(feature = "polkadot-native")]
 			"polkadot-local" => Box::new(service::chain_spec::polkadot_local_testnet_config()?),
 			#[cfg(feature = "polkadot-native")]
 			"polkadot-staging" => Box::new(service::chain_spec::polkadot_staging_testnet_config()?),
-			"rococo" => Box::new(service::chain_spec::rococo_config()?),
 			#[cfg(feature = "rococo-native")]
 			"rococo-dev" => Box::new(service::chain_spec::rococo_development_config()?),
 			#[cfg(feature = "rococo-native")]
@@ -109,7 +106,6 @@
 			#[cfg(not(feature = "rococo-native"))]
 			name if name.starts_with("rococo-") && !name.ends_with(".json") =>
 				Err(format!("`{}` only supported with `rococo-native` feature enabled.", name))?,
-			"westend" => Box::new(service::chain_spec::westend_config()?),
 			#[cfg(feature = "westend-native")]
 			"westend-dev" => Box::new(service::chain_spec::westend_development_config()?),
 			#[cfg(feature = "westend-native")]
@@ -119,7 +115,6 @@
 			#[cfg(not(feature = "westend-native"))]
 			name if name.starts_with("westend-") && !name.ends_with(".json") =>
 				Err(format!("`{}` only supported with `westend-native` feature enabled.", name))?,
-			"wococo" => Box::new(service::chain_spec::wococo_config()?),
 			#[cfg(feature = "rococo-native")]
 			"wococo-dev" => Box::new(service::chain_spec::wococo_development_config()?),
 			#[cfg(feature = "rococo-native")]
@@ -492,6 +487,7 @@
 			let runner = cli.create_runner(cmd)?;
 			let chain_spec = &runner.config().chain_spec;
 
+			#[allow(clippy::complexity)]
 			match cmd {
 				BenchmarkCmd::Storage(cmd) => runner.sync_run(|mut config| {
 					let (client, backend, _, _) = service::new_chain_ops(&mut config, None)?;
