--- pallets/assets/src/tests_fungibles.rs.tmp
+++ pallets/assets/src/tests_fungibles.rs
@@ -26,10 +26,13 @@
 
 #[test]
 fn fungibles_mutate_trait_should_work() {
-	ExtBuilder::default().build().execute_with(|| {
-		assert_ok!(<Tokens as fungibles::Mutate<_>>::mint_into(DOT, &ALICE, 10));
-		assert_eq!(<Tokens as fungibles::Mutate<_>>::burn_from(DOT, &ALICE, 8), Ok(8));
-	});
+	ExtBuilder::default()
+		.balances(vec![(ALICE, DOT, 100)])
+		.build()
+		.execute_with(|| {
+			assert_ok!(<Tokens as fungibles::Mutate<_>>::mint_into(DOT, &ALICE, 10));
+			assert_eq!(<Tokens as fungibles::Mutate<_>>::burn_from(DOT, &ALICE, 8), Ok(8));
+		});
 }
 
 #[test]
