--- node/.tmp/original/Cargo.toml
+++ node/.tmp/migrate/Cargo.toml
@@ -17,7 +17,7 @@
 serde = { version = "1.0.152", features = ["derive"] }
 
 # Local
-parachain-template-runtime = { git = "https://github.com/substrate-developer-hub/substrate-parachain-template", rev = "6cde4f130cc41287ef2ed5a00093991197f3c1a3" }
+parachain-template-runtime = { path = "../runtime" }
 
 # Substrate
 frame-benchmarking = { git = "https://github.com/paritytech/substrate", "branch" = "polkadot-v0.9.37" }
@@ -55,20 +55,33 @@
 try-runtime-cli = { git = "https://github.com/paritytech/substrate", optional = true, "branch" = "polkadot-v0.9.37" }
 
 # Polkadot
-polkadot-cli = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.37" }
+polkadot-cli = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.37", features = ["rococo-native", "fast-runtime"] }
 polkadot-primitives = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.37" }
 polkadot-service = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.37" }
 xcm = { git = "https://github.com/paritytech/polkadot", default-features = false, branch = "release-v0.9.37" }
 
 # Cumulus
-cumulus-client-cli = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.37" }
-cumulus-client-consensus-aura = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.37" }
-cumulus-client-consensus-common = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.37" }
-cumulus-client-network = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.37" }
-cumulus-client-service = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.37" }
-cumulus-primitives-core = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.37" }
-cumulus-primitives-parachain-inherent = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.37" }
-cumulus-relay-chain-interface = { git = "https://github.com/paritytech/cumulus.git", branch = "polkadot-v0.9.37" }
+cumulus-client-cli = { path = "../cumulus/client/cli" }
+cumulus-client-consensus-aura = { path = "../cumulus/client/consensus/aura" }
+cumulus-client-consensus-common = { path = "../cumulus/client/consensus/common" }
+cumulus-client-network = { path = "../cumulus/client/network" }
+cumulus-client-service = { path = "../cumulus/client/service" }
+cumulus-primitives-core = { git = "https://github.com/paritytech/cumulus", default-features = false, branch = "polkadot-v0.9.37" }
+cumulus-primitives-parachain-inherent = { path = "../cumulus/primitives/parachain-inherent" }
+cumulus-relay-chain-interface = { path = "../cumulus/client/relay-chain-interface" }
+
+# Frontier dependencies
+fc-consensus = { path = "../frontier/client/consensus" }
+fc-db = { path = "../frontier/client/db" }
+fc-mapping-sync = { path = "../frontier/client/mapping-sync" }
+fc-rpc = { path = "../frontier/client/rpc" }
+fc-rpc-core = { path = "../frontier/client/rpc-core" }
+fp-consensus = { path = "../frontier/primitives/consensus" }
+fp-evm = { path = "../frontier/primitives/evm" }
+fp-rpc = { path = "../frontier/primitives/rpc" }
+fp-storage = { path = "../frontier/primitives/storage" }
+pallet-ethereum = { path = "../frontier/frame/ethereum" }
+pallet-evm = { path = "../frontier/frame/evm" }
 
 [build-dependencies]
 substrate-build-script-utils = { git = "https://github.com/paritytech/substrate", "branch" = "polkadot-v0.9.37" }
@@ -84,3 +97,5 @@
   "try-runtime-cli/try-runtime",
   "parachain-template-runtime/try-runtime",
 ]
+
+fast-runtime = ["parachain-template-runtime/fast-runtime", "polkadot-cli/fast-runtime"]
