--- node/relaychain-rio-service/Cargo.toml.tmp
+++ node/relaychain-rio-service/Cargo.toml
@@ -1,5 +1,5 @@
 [package]
-name = "polkadot-service"
+name = "relaychain-rio-service"
 version = "0.9.22"
 authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2021"
@@ -76,7 +76,7 @@
 # Polkadot
 polkadot-node-core-parachains-inherent = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22" }
 polkadot-overseer = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22" }
-polkadot-client = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", default-features = false, optional = true }
+polkadot-client = { package = "relaychain-rio-client", path = "../relaychain-rio-client", default-features = false, optional = true }
 polkadot-parachain = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22" }
 polkadot-primitives = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22" }
 polkadot-node-primitives = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22" }
@@ -87,13 +87,13 @@
 polkadot-node-network-protocol = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22" }
 
 # Polkadot Runtime Constants
-polkadot-runtime-constants = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
+polkadot-runtime-constants = { package = "relaychain-rio-runtime-constants", path = "../../runtime/relaychain-rio/constants", optional = true }
 kusama-runtime-constants = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
 rococo-runtime-constants = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
 westend-runtime-constants = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
 
 # Polkadot Runtimes
-polkadot-runtime = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
+polkadot-runtime = { package = "relaychain-rio-runtime", path = "../../runtime/relaychain-rio", optional = true }
 kusama-runtime = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
 westend-runtime = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
 rococo-runtime = { git = "https://github.com/paritytech/polkadot", branch = "release-v0.9.22", optional = true }
